substratevm/src/com.oracle.svm.agent/src/com/oracle/svm/agent/NativeImageTracingAgent.java

```diff
@@ -88,53 +70,58 @@
 import com.oracle.svm.configure.config.ConfigurationSet;
 import com.oracle.svm.configure.filters.FilterConfigurationParser;
 import com.oracle.svm.configure.filters.RuleNode;
+import com.oracle.svm.configure.json.JsonPrintable;
 import com.oracle.svm.configure.json.JsonWriter;
 import com.oracle.svm.configure.trace.AccessAdvisor;
 import com.oracle.svm.configure.trace.TraceProcessor;
 import com.oracle.svm.core.FallbackExecutor;
 import com.oracle.svm.core.SubstrateUtil;
-import com.oracle.svm.core.c.function.CEntryPointOptions;
-import com.oracle.svm.core.c.function.CEntryPointSetup;
 import com.oracle.svm.core.configure.ConfigurationFiles;
 import com.oracle.svm.driver.NativeImage;
 import com.oracle.svm.jni.nativeapi.JNIEnvironment;
-import com.oracle.svm.jni.nativeapi.JNIErrors;
 import com.oracle.svm.jni.nativeapi.JNIJavaVM;
 import com.oracle.svm.jni.nativeapi.JNIObjectHandle;
-import com.oracle.svm.jni.nativeapi.JNIVersion;
+import org.graalvm.nativeimage.hosted.Feature;
 
-public final class Agent {
-    public static final String AGENT_NAME = "native-image-agent";
+public final class NativeImageTracingAgent extends JvmtiAgentBase<NativeImageAgentJNIHandleSet> {
```

Comment:
I'd rather keep the name `NativeImageAgent`, which is also consistent with the name of the build artifact. Tracing is only part of the agent functionality, the other major part being the restriction mode.

Reply:
I agree with you, I have changed the name.